tests/publish/test_publish.py

```diff
@@ -238,6 +255,46 @@ def test_repo_publish_cache_cleanup(command_tester):
     assert fake_publish.udcache_client.flushed_repos == ["repo1"]
 
 
+def test_repo_publish_cache_cleanup_skip_ud(command_tester):
+    """publishes the repo provided, doesn't clean up UD cache if repo missing eng ID"""
+    with FakePublish() as fake_publish:
+        fake_pulp = fake_publish.pulp_client_controller
+        _add_repo(fake_pulp)
+
+        command_tester.test(
+            fake_publish.main,
+            [
+                "test-publish",
+                "--pulp-url",
+                "https://pulp.example.com",
+                "--fastpurge-host",
+                "fakehost-xxx.example.net",
+                "--fastpurge-client-secret",
+                "abcdef",
+                "--fastpurge-client-token",
+                "efg",
+                "--fastpurge-access-token",
+                "tok",
+                "--fastpurge-root-url",
+                "https://cdn.example.com/",
+                "--udcache-url",
+                "https://ud.example.com/",
+                "--repo-ids",
+                "repo4",
+            ],
+        )
+
+    # pulp repo is published
+    assert [hist.repository.id for hist in fake_pulp.publish_history] == ["repo4"]
+    # flushed the urls
+    assert sorted(fake_publish.fastpurge_client.purged_urls) == [
+        "https://cdn.example.com/content/unit/4/client/mutable1",
+        "https://cdn.example.com/content/unit/4/client/mutable2",
+    ]
+    # did not flush the UD object
+    assert fake_publish.udcache_client.flushed_repos == []
```

Comment:
Good to have this test case include a repo with eng ID as well so that it covers the all the cases i.e. this or any change in future makes the code skip only the relevant repo without an eng ID only

Reply:
Hmm. Yeah, could do that. I almost included this new repo in an existing test to achieve that effect. Probably helpful to see both in a single case, as you say, so I'll add.